edi_oca/models/edi_exchange_record.py

```diff
@@ -116,12 +129,30 @@ def _compute_related_name(self):
             related_record = rec.record
             rec.related_name = related_record.display_name if related_record else ""
 
-    @api.depends("model", "type_id")
+    @api.depends("parent_id", "real_exchange_file")
+    def _compute_exchange_file(self):
+        for rec in self:
+            if rec.real_exchange_file:
+                rec.exchange_file = rec.real_exchange_file
+            elif rec.parent_id:
+                rec.exchange_file = rec.parent_id.exchange_file
+            else:
+                rec.exchange_file = False
+            rec.has_exchange_file = True if rec.exchange_file else False
+
+    def _inverse_exchange_file(self):
+        for rec in self:
+            rec.real_exchange_file = rec.exchange_file
+            rec.has_exchange_file = True
+
+    @api.depends("model", "type_id", "parent_id")
     def _compute_exchange_filename(self):
         for rec in self:
-            if not rec.type_id:
+            if rec.exchange_filename:
                 continue
-            if not rec.exchange_filename:
+            if not rec.real_exchange_file and rec.parent_id.exchange_file:
```

Comment:
IMO, this should only happen in some kind of backends or backend types, so I would add a condition there.

Reply:
I dont know, I can have the same thing that we have for the exchange file but it has constraints (harder to search on it, ...) and the db footprint is reasonable (to me, i may be wrong)